cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/loadbalancing/DescribeLoadBalancers.java

```diff
@@ -0,0 +1,159 @@
+import com.hp.oo.sdk.content.annotations.Response;
+import com.hp.oo.sdk.content.plugin.ActionMetadata.MatchType;
+import com.hp.oo.sdk.content.plugin.ActionMetadata.ResponseType;
+import io.cloudslang.content.amazon.entities.inputs.CommonInputs;
+import io.cloudslang.content.amazon.entities.inputs.LoadBalancerInputs;
+import io.cloudslang.content.amazon.execute.QueryApiExecutor;
+import io.cloudslang.content.amazon.utils.ExceptionProcessor;
+import io.cloudslang.content.constants.ReturnCodes;
+
+import java.util.Map;
+
+import static io.cloudslang.content.amazon.entities.constants.Constants.Apis.LOAD_BALANCING_API;
+import static io.cloudslang.content.amazon.entities.constants.Constants.AwsParams.HTTP_CLIENT_METHOD_GET;
+import static io.cloudslang.content.amazon.entities.constants.Constants.DefaultApiVersion.LOAD_BALANCER_DEFAULT_API_VERSION;
+import static io.cloudslang.content.amazon.entities.constants.Constants.LoadBalancingQueryApiActions.DESCRIBE_LOAD_BALANCERS;
+import static io.cloudslang.content.amazon.entities.constants.Constants.Miscellaneous.EMPTY;
+import static io.cloudslang.content.amazon.entities.constants.Inputs.CommonInputs.*;
+import static io.cloudslang.content.amazon.entities.constants.Inputs.LoadBalancerInputs.*;
+import static io.cloudslang.content.amazon.utils.InputsUtil.getDefaultStringInput;
+import static io.cloudslang.content.constants.OutputNames.EXCEPTION;
+import static io.cloudslang.content.constants.OutputNames.RETURN_CODE;
+import static io.cloudslang.content.constants.OutputNames.RETURN_RESULT;
+import static io.cloudslang.content.constants.ResponseNames.FAILURE;
+import static io.cloudslang.content.constants.ResponseNames.SUCCESS;
+
+/**
+ * Created by TusaM
+ * 3/8/2017.
+ */
+public class DescribeLoadBalancers {
+    /**
+     * Describes the specified Application Load Balancers or all of your Application Load Balancers.
+     * Note: To describe the listeners for a load balancer, use DescribeListeners:
+     * http://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_DescribeListeners.html
+     * To describe the attributes for a load balancer, use DescribeLoadBalancerAttributes:
+     * http://docs.aws.amazon.com/elasticloadbalancing/latest/APIReference/API_DescribeLoadBalancerAttributes.html
+     *
+     * @param endpoint          Optional - Endpoint to which request will be sent.
+     *                          Default: "https://elasticloadbalancing.amazonaws.com"
+     * @param identity          ID of the secret access key associated with your Amazon AWS or IAM account.
+     *                          Example: "AKIAIOSFODNN7EXAMPLE"
+     * @param credential        Secret access key associated with your Amazon AWS or IAM account.
+     *                          Example: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+     * @param proxyHost         Optional - proxy server used to connect to Amazon API. If empty no proxy will be used.
+     *                          Default: ""
```

Comment:
empty defaults are not needed ... they are redundant and in some cases they are not set at all if "isEmpty" ...  same goes for all inputs with empty default